pkg/config/config.go

```diff
@@ -0,0 +1,119 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/OctopusDeploy/cli/pkg/constants"
+	"github.com/spf13/viper"
+)
+
+const configName = "cli_config"
+const defaultConfigFileType = "json"
+const appData = "AppData"
+
+// var configPath string
+
+func Setup() {
+	viper.SetDefault(constants.ConfigHost, "")
+	viper.SetDefault(constants.ConfigApiKey, "")
+	viper.SetDefault(constants.ConfigSpace, "")
+	viper.SetDefault(constants.ConfigNoPrompt, false)
+	//	viper.SetDefault(constants.ConfigProxyUrl, "")
+	//	viper.SetDefault(constants.ConfigShowOctopus, true)
+	viper.SetDefault(constants.ConfigOutputFormat, "table")
+
+	viper.SetConfigName(configName)
+	viper.SetConfigType(defaultConfigFileType)
+
+	if runtime.GOOS == "windows" {
+		viper.SetDefault(constants.ConfigEditor, "notepad")
+	} else { // unix
+		viper.SetDefault(constants.ConfigEditor, "nano")
+	}
+	getConfigPath()
+
+	if err := viper.ReadInConfig(); err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+			// Do nothing, config file will be created on `config set` cmd
+			// This is to avoid issues with CI tools that may not have access
+			// to the file system
+		} else {
+			// Config file was found but something is wrong
+			panic(fmt.Errorf("fatal error reading config file: %w", err))
+		}
+	}
+}
+
+func SetupEnv() {
+	viper.BindEnv(constants.ConfigApiKey, constants.EnvOctopusApiKey)
+	viper.BindEnv(constants.ConfigHost, constants.EnvOctopusHost)
+	viper.BindEnv(constants.ConfigSpace, constants.EnvOctopusSpace)
+	viper.BindEnv(constants.ConfigEditor, constants.EnvEditor, constants.EnvVisual)
```

Comment:
q: if both EDITOR and VISUAL are set, which one wins here? Might be useful to add a comment in the code explaining this

Reply:
They will take precedence in the specified order, will leave a comment.